deltachat-jsonrpc/src/api/types/calls.rs

```diff
@@ -51,3 +52,82 @@ impl JsonrpcCallInfo {
         })
     }
 }
+
+#[derive(Serialize, TypeDef, schemars::JsonSchema)]
+#[serde(rename = "CallState", tag = "kind")]
+pub enum JsonrpcCallState {
+    /// Fresh incoming call that should be still ringing.
+    IncomingRinging,
+
+    /// Incoming call that was not picked up within a timeout
+    /// or was explicitly ended by the caller before we picked up.
+    IncomingMissed,
+
+    /// Incoming call that was explicitly ended on our side
+    /// before picking up.
+    IncomingDeclined,
+
+    /// Active incoming call with duration in seconds.
+    IncomingActive {
+        /// Call duration in seconds.
+        duration: i64,
+    },
+
+    /// Completed incoming call with duration in seconds.
+    IncomingCompleted {
+        /// Call duration in seconds.
+        duration: i64,
+    },
+
+    /// Fresh outgoing call that is still ringing.
+    ///
+    /// There is no separate state for the call
+    /// that has been dialled but not ringing on the other side yet
+    /// as we don't know whether the other side received our call.
+    OutgoingRinging,
+
+    /// Outgoing call that has been declined by the other side.
+    OutgoingDeclined,
+
+    /// Outgoing call that has been cancelled on our side
+    /// before receiving a response.
+    OutgoingCancelled,
+
+    /// Active outgoing call.
+    OutgoingActive {
+        /// Call duration in seconds.
+        duration: i64,
```

Comment:
I don't think this is needed, the UI will not be calling this API over and over every second to update the time anyways, if anything, call start time is what would have made sense, I recommend simply not to provide anything in this state for now